internal/pkg/template/env.go

```diff
@@ -131,10 +131,11 @@ func (e *EnvOpts) HasImportedCerts() bool {
 
 // HTTPConfig represents configuration for a Load Balancer.
 type HTTPConfig struct {
-	CIDRPrefixListIDs []string
-	ImportedCertARNs  []string
-	CustomALBSubnets  []string
-	ELBAccessLogs     *ELBAccessLogs
+	CIDRPrefixListIDs  []string
+	ImportedCertARNs   []string
+	CustomALBSubnets   []string
+	ELBAccessLogs      *ELBAccessLogs
+	PublicALBSourceIPs []string
```

Comment:
Hmm my understanding is that this type is for any type of LB. Whereas, the new field is strictly for public ALBs.

What if we create a new type here:
```go
type PublicHTTPConfig struct {
   HTTPConfig
   PublicALBSourceIPs []string
}
func (cfg *PublicHTTPConfig) HasCustomIngress() bool {
   return len(cfg.PublicALBSourceIPs) > 0 && len(CIDRPrefixListIDs) >0 
}
```